packages/starlight/utils/navigation.ts

```diff
@@ -79,13 +79,16 @@ function groupFromAutogenerateConfig(
 ): Group {
 	const { collapsed: subgroupCollapsed, directory } = item.autogenerate;
 	const localeDir = locale ? locale + '/' + directory : directory;
-	const dirDocs = routes.filter(
-		(doc) =>
-			// Match against `foo.md` or `foo/index.md`.
-			stripExtension(doc.id) === localeDir ||
-			// Match against `foo/anything/else.md`.
-			doc.id.startsWith(localeDir + '/')
-	);
+	const dirDocs = routes
+		.filter(
+			(doc) =>
+				// Match against `foo.md` or `foo/index.md`.
+				stripExtension(doc.id) === localeDir ||
+				// Match against `foo/anything/else.md`.
+				doc.id.startsWith(localeDir + '/')
+		)
+		// Remove any entries that should be hidden
+		.filter((doc) => !doc.entry.data.sidebar.hidden);
```

Comment:
Any specific reason to not do that in the `filter` right above?

Reply:
At first I did that to make it more readable instead of nesting logic, but I think it'd be better to include it all so that the array isn't iterated over twice.